File: fo/src/types/product.ts
// 상품 타입
export interface Product {
  productId: number;
  name: string;
  price: number;
}

// 상품 생성 요청 타입
export interface CreateProductRequest {
  name: string;
  price: number;
}

// 상품 수정 요청 타입
export interface UpdateProductRequest {
  name?: string;
  price?: number;
}

// 상품 검색 조건 타입
export interface ProductSearchCondition {
  name?: string;
  minPrice?: number;
  maxPrice?: number;
}

File: api/src/main/java/com/vibe/pay/backend/product/Product.java
package com.vibe.pay.backend.product;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * @author Claude
 * @version 1.0
 * @since 2025-01-28
 */
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    private Long productId;
    private String name;
    private Double price;
}
